x/auth/client/rest/query.go

```diff
@@ -88,7 +88,7 @@ func QueryBalancesRequestHandlerFn(
 
 		// the query will return empty if there is no data for this account
 		if len(res) == 0 {
-			w.WriteHeader(http.StatusNoContent)
+			w.WriteHeader(http.StatusOK)
```

Comment:
curious if this function is a duplicate from the one above ?

Reply:
Most of it is duplicated, only change is response. Eg. for `/auth/accounts/{address}` we are returning `Account{}` which in case `/bank/balances/{address}` we are returning `sdk.Coins`

The balance endpoint does not make sense for me, it should be `/auth/accounts/{address}/balances` and the same function should check the endpoint and return corresponding model. But this is a API breaking change so maybe it should be a separated issue for code cleanup :)

Reply:
I agree it doesn’t make much sense. Let’s open an issue for it